example_api/main.cpp

```diff
@@ -159,4 +159,48 @@ int main()
 	ttdcursor.AddMemoryWatchpoint(&data);
 	ttdcursor.ReplayForward(&replayrez, last, -1);
 	printf("%llx\n", ttdcursor.GetProgramCounter());
+
+	// Exemple of timeline
+	std::cout << "Timeline" << std::endl;
+	auto moduleLoaded = ttdengine.GetModuleLoadedEvent();
+	auto moduleUnloaded = ttdengine.GetModuleUnloadedEvent();
+	auto threadCreate = ttdengine.GetThreadCreatedEvent();
+	auto threadTerminate = ttdengine.GetThreadTerminatedEvent();
+
+	auto itModuleLoaded = moduleLoaded.begin();
+	auto itModuleUnloaded = moduleUnloaded.begin();
+	auto itThreadCreate = threadCreate.begin();
+	auto itThreadTerminate = threadTerminate.begin();
+
+	while (itModuleLoaded != moduleLoaded.end() || itModuleUnloaded != moduleUnloaded.end() || itThreadCreate != threadCreate.end() || itThreadTerminate != threadTerminate.end())
+	{
+		TTD::Position moduleLoadedPosition = (itModuleLoaded == moduleLoaded.end()) ? TTD::MAX : itModuleLoaded->pos;
+		TTD::Position moduleUnloadedPosition = (itModuleUnloaded == moduleUnloaded.end()) ? TTD::MAX : itModuleUnloaded->pos;
+		TTD::Position threadCreatePosition = (itThreadCreate == threadCreate.end()) ? TTD::MAX : itThreadCreate->pos;
+		TTD::Position threadTerminatePosition = (itThreadTerminate == threadTerminate.end()) ? TTD::MAX : itThreadTerminate->pos;
+
+ 		if (moduleLoadedPosition < moduleUnloadedPosition && moduleLoadedPosition < threadCreatePosition && moduleLoadedPosition < threadTerminatePosition)
+		{
+			std::wcout << "Module Loaded at " << std::hex << itModuleLoaded->pos.Major << ":" << std::hex << itModuleLoaded->pos.Minor << " : " << std::wstring(itModuleLoaded->info->path) << std::endl;
```

Comment:
Maybe we should add a position display function to avoid such code

Reply:
Yes I'm afraid to write this code but as the two object are different, i didn't find a better way to do it...